nautobot/extras/models/datasources.py

```diff
@@ -167,3 +173,51 @@ def sync(self, user, dry_run=False):
         if dry_run:
             return enqueue_git_repository_diff_origin_and_local(self, user)
         return enqueue_pull_git_repository_and_refresh_data(self, user)
+
+    def clone(self, path=None, branch=None, head=None):
+        """
+        Perform a shallow clone of the Git repository in a temporary directory.
+
+        Args:
+            path (str): Optional The absolute path to clone the repository to. If not specified, the directory will be created in the /tmp directory.
+            branch (str): Optional The branch to checkout. If not set, the GitRepository.branch will be used.
+            head (str): Optional Git commit hash to check out instead of pulling branch latest.
+
+        Returns:
+            Return the absolute path of the cloned repo if clone was successful
+        """
+
+        absolute_path_name = tempfile.mkdtemp(dir=path)
```

Comment:
Probably need a try/except here in case the user tries to create in a path they don't have write access to.